### PR TITLE
Update Core Tools for E2E to 4.7.0

### PR DESCRIPTION
**Pull request by andystaples**
# Summary
## What changed?
Updates the Core Tools version used by the E2E tests to the latest (4.7.0)

Also contains the following E2E test improvements: 
- Updates the Python version used for e2e tests to 3.13
- Updates the Node version used to install azurite and run Node e2e tests to Node 20
- Uses a Python virtual environment around the tests to avoid altering the global Python env
- Updates old dependencies in Node package-lock

## Why is this change needed?
Compatibility with Python 3.13, test against more modern Host

## Issues / work items
- Resolves #
- Related #

---

# Project checklist
- [x] Documentation changes are not required
  - [ ] Otherwise: Documentation PR is ready to merge and referenced in `pending_docs.md`
- [x] Release notes are not required for the next release
  - [ ] Otherwise: Notes added to `release_notes.md`
- [x] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [x] All required tests have been added/updated (unit tests, E2E tests)
- [x] No extra work is required to be leveraged by OutOfProc SDKs
  - [ ] Otherwise: Work tracked here: #issue_or_pr_in_each_sdk
- x ] No change to the version of the `WebJobs.Extensions.DurableTask` package
  - [ ] Otherwise: Major/minor updates are reflected in `/src/Worker.Extensions.DurableTask/AssemblyInfo.cs`
- [x] No EventIds were added to `EventSource` logs
  - [ ] Otherwise: Ensure EventIds are within the supported range in the existing Windows infrastructure (validate via deployed telemetry). If needed, extend the range via a PR such as https://msazure.visualstudio.com/One/_git/AAPT-Antares-Websites/pullrequest/7463263?_a=files
- [ ] This change should be added to the `v2.x` branch
  - [x] Otherwise: This change applies exclusively to `WebJobs.Extensions.DurableTask` v3.x and will be retained only in the `dev` and `main` branches
- [ ] Breaking change?
  - [ ] If yes:
    - Impact:
    - Migration guidance:
---

# AI-assisted code disclosure (required)
## Was an AI tool used? (select one)
- [x] No
- [ ] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [ ] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s):
- AI-assisted areas/files:
- What you changed after AI output:

AI verification (required if AI was used):
- [ ] I understand the code and can explain it
- [ ] I verified referenced APIs/types exist and are correct
- [ ] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [ ] I reviewed concurrency/async behavior
- [ ] I checked for unintended breaking or behavior changes

### CODE REVIEW

File: test/e2e/Tests/build-e2e-test.ps1 [hunk 1]
```diff
@@ -120,12 +127,24 @@ else
...
+
 function InstallExtensionAndBuildTestApp($testAppDir) {
     Write-Host "Building test app $testAppDir"
-    Set-Location $testAppDir
+    Push-Location $testAppDir
+    try {
 
     Write-Host "Removing cached WebJobs extension versions from nuget cache, if exists"
-    $cachedVersionFolders = Get-ChildItem -Path (Join-Path $LocalNugetCacheDirectory "microsoft.azure.webjobs.extensions.durabletask") -Directory -ErrorAction Continue
+    $cachedVersionFolders = Get-ChildItem -Path (Join-Path $LocalNugetCacheDirectory "microsoft.azure.webjobs.extensions.durabletask") -Directory -ErrorAction SilentlyContinue
```
Comment (bachuv) [inline test/e2e/Tests/build-e2e-test.ps1:147]: Should we keep the `Continue` here so we can see the error? From what I understand, `SilentlyContinue` doesn't show us an error if there is one. Let me know if the behavior is actually different.

File: test/e2e/Tests/build-e2e-test.ps1 [hunk 2]
```diff
@@ -257,7 +302,9 @@ if ($StartMSSqlContainer)
       Write-Warning "No MSSQL_SA_PASSWORD environment variable found! Skipping SQL Server container startup."
     }
   }
-  StartMSSQLContainer $MSSQLpwd
+  if ($MSSQLpwd) {
+    StartMSSQLContainer $MSSQLpwd
+  }
```
Comment (bachuv) [inline test/e2e/Tests/build-e2e-test.ps1:307]: Do we need this check? Seems like `$MSSQLpwd` will be set at this point.

File: test/e2e/Tests/build-e2e-test.ps1 (hunk 1 above)
Comment (andystaples) [reply test/e2e/Tests/build-e2e-test.ps1:147]: Yeah - SilentlyContinue is preferred here, previously with -Continue it was spilling red output into our CI on every run clogging the output. If Get-ChildItem fails, that means the directory doesn't exist, so NBD to just skip and move on

File: test/e2e/Tests/build-e2e-test.ps1 (hunk 2 above)
Comment (andystaples) [reply test/e2e/Tests/build-e2e-test.ps1:307]: If the caller doesn't pass one as a script arg, and $env:MSSQL_SA_PASSWORD is also empty, we write the warning on the lines above but weren't previously skipping as the log would indicate. This check corrects that

**Review by bachuv — APPROVED** [review]
